tests/unit/logic/vote.spec.js

```diff
@@ -0,0 +1,1012 @@
+				it('should throw an exception', function () {
+					execMethod = function () { vote.objectNormalize(trs_fail_5) }
+					expect(execMethod).to.throw('Array items are not unique')
+				})
+			})
+
+			context('Fail 6: if trs.asset.votes is greater than constants.maxVotesPerTransaction', function () {
+				it('should throw an exception', function () {
+					execMethod = function () { vote.objectNormalize(trs_fail_6) }
+					expect(execMethod).to.throw('Array is too long')
+				})
+			})
+
+			context('Fail 7: if trs.asset has additional properties', function () {
+				it('should throw an exception', function () {
+					execMethod = function () { vote.objectNormalize(trs_fail_7) }
+					expect(execMethod).to.throw('Additional properties not allowed')
+				})
+			})
+
+			context('Fail 8: if trs.asset.votes items has a wrong format', function () {
+				it('should throw an exception', function () {
+					execMethod = function () { vote.objectNormalize(trs_fail_8) }
+					expect(execMethod).to.throw('String does not match pattern')
+					expect(execMethod).to.throw('Expected type string but found type integer')
+				})
+			})
+
+			context('if everything is ok', function () {
+				it('should return the same transaction object', function () {
+					var result = vote.objectNormalize(trs_success)
+					expect(result).to.deep.equal(trs_success)
+				})
+			})
+		})
+
+		describe('dbRead()', function () {
+
+			var raw_success, raw_fail_1, raw_fail_2, raw_fail_3, raw_fail_4, raw_fail_5
+
+			beforeEach(function () {
+				raw_fail_1 = {v_votes: undefined}
+				raw_fail_2 = {v_votes: null}
+				raw_fail_3 = {v_votes: false}
+				raw_fail_4 = {v_votes: ''}
+				raw_fail_5 = {v_votes: []}
+				raw_success = {v_votes: '1,3,4'}
+			})
+
+			context('Fail 1: If v_votes is undefined', function () {
+				it('should return null', function () {
+					var result = vote.dbRead(raw_fail_1)
+					expect(result).to.equals(null)
+				})
+			})
+
+			context('Fail 2: If v_votes is null', function () {
+				it('should return null', function () {
+					var result = vote.dbRead(raw_fail_2)
+					expect(result).to.equals(null)
+				})
+			})
+
+			context('Fail 3: If v_votes is false', function () {
+				it('should return null', function () {
+					var result = vote.dbRead(raw_fail_3)
+					expect(result).to.equals(null)
+				})
+			})
+
+			context('Fail 4: If v_votes is an empty string', function () {
+				it('should return null', function () {
+					var result = vote.dbRead(raw_fail_4)
+					expect(result).to.equals(null)
+				})
+			})
+
+			context('Fail 5: If v_votes is not a string', function () {
+				it('should return null', function () {
+					var result = vote.dbRead(raw_fail_5)
+					expect(result).to.equals(null)
+				})
+			})
+
+			context('if everything is ok', function () {
+				it('should return an object with votes as property in Array format', function () {
+					var result = vote.dbRead(raw_success)
+					expect(result).to.have.deep.property('votes')
```

Comment:
I think that the next expect is enough, or?

Reply:
I added the second one for to verify that the values received are correct.